keyboards/ergodox_ez/keymaps/mac_win_german_programmer/keymap.c

```diff
@@ -0,0 +1,357 @@
+
+      // Right
+      DE_RBRC,KC_6,KC_7,KC_8,KC_9,KC_0,DE_LESS,
+      DE_RCBR,DE_Z,KC_U,KC_I,KC_O,KC_P,NO_SLSH,
+      KC_H,KC_J,KC_K,KC_L,LT(ALL_MOUSE,WIN_DE_QUESTION),OSL(WIN_UMLAUT),NO_RPRN,
+      KC_N,KC_M,KC_COMMA,KC_DOT,NO_MINS,KC_RSHIFT,
+      KC_UP,KC_DOWN,KC_NO,KC_NONUS_HASH,KC_RALT,
+      // BLOCK
+      KC_LALT,CTL_T(KC_ESCAPE),KC_PGUP,KC_PGDOWN,KC_ENTER,KC_SPACE
+    ),
+
+    [WIN_SPECIAL] =  KEYMAP(
+      // Left
+      KC_ESCAPE,KC_F1,KC_F2,KC_F3,KC_F4,KC_F5,KC_NO,
+      KC_NO,DE_AT,TO(WIN_ALPHA),KC_NO,KC_NO,KC_NO,KC_NO,
+      KC_NO,KC_HASH,KC_DLR,KC_LPRN,KC_RPRN,KC_GRAVE,
+      KC_LSHIFT,KC_PERC,KC_CIRC,KC_LBRACKET,KC_RBRACKET,KC_TILD,KC_NO,
+      KC_NO,KC_LALT,KC_NO,KC_NO,KC_NO,
+      // BLOCK
+      RGB_MOD,KC_NO,KC_NO,RGB_VAD,RGB_VAI,KC_NO,
+
+      // Right
+      KC_NO,KC_F6,KC_F7,KC_F8,KC_F9,WIN_DE_EQUAL,KC_KP_ASTERISK,
+      KC_NO,KC_NO,KC_NO,KC_NO,KC_NO,KC_NONUS_HASH,NO_SLSH,
+      KC_LEFT,KC_DOWN,KC_UP,KC_RIGHT,MAC_DE_PIPE,KC_KP_PLUS,
+      KC_NO,KC_NO,TO(MACOS_ALPHA),KC_NO,KC_NO,KC_NONUS_BSLASH,KC_KP_MINUS,
+      KC_NO,KC_NO,KC_NO,DE_TILD,WIN_DE_GRAVE,
+      // BLOCK
+      RGB_TOG,RGB_SLD,KC_NO,KC_NO,RGB_HUD,RGB_HUI
+    ),
+
+
+    [WIN_UMLAUT] = KEYMAP(
+      // Left
+      KC_NO,KC_NO,KC_NO,KC_NO,KC_NO,KC_NO,KC_NO, // 7
+      KC_NO,DE_AT,KC_NO,DE_EURO,KC_NO,KC_NO,KC_NO, // 7
+      KC_NO,DE_AE,DE_SS,KC_NO,KC_NO,KC_NO, // 6
+      KC_LSHIFT,KC_NO,KC_NO,KC_NO,KC_NO,KC_NO,KC_NO, // 7
+      KC_NO,KC_LALT,KC_NO,KC_NO,KC_NO, // 5
+      // BLOCK
+      KC_NO,KC_NO,KC_NO,KC_NO,KC_NO,KC_NO, // 6
+
+      // Right
+      KC_NO,KC_NO,KC_NO,KC_NO,KC_NO,KC_NO,KC_NO, // 7
+      KC_NO,DE_Z,DE_UE,KC_NO,DE_OE,KC_NO,KC_NO, // 7
+      KC_NO,KC_NO,KC_NO,KC_NO,KC_NO,KC_NO, // 6
+      KC_NO,KC_NO,KC_NO,KC_NO,KC_NO,KC_NO,KC_RSHIFT, // 7
+      KC_NO,KC_NO,KC_NO,KC_NO,KC_NO,  // 5
+      // BLOCK
+      KC_NO,KC_NO,KC_NO,KC_NO,KC_NO,KC_NO // 6
+    ),
+
+    [WIN_EMPTY1] = EMPTY_LAYER,
+    [WIN_EMPTY2] = EMPTY_LAYER,
+
+  [ALL_MOUSE] = KEYMAP(
+    // Left
+    KC_NO,KC_NO,KC_NO,KC_NO,KC_NO,KC_NO,KC_NO,
+    KC_NO,KC_NO,KC_MS_UP,KC_NO,KC_NO,KC_NO,KC_NO,
+    KC_NO,KC_MS_LEFT,KC_MS_DOWN,KC_MS_RIGHT,KC_MS_RIGHT,KC_NO,
+    KC_NO,KC_NO,KC_NO,KC_NO,KC_NO,KC_NO,KC_NO,
+    KC_NO,KC_LALT,KC_NO,KC_NO,KC_NO,KC_NO,KC_NO,
+
+    // BLOCK
+    KC_NO,KC_MS_BTN1,KC_MS_BTN2,KC_NO,KC_NO,KC_NO,
+
+    // Right
+    KC_NO,KC_NO,KC_NO,KC_NO,KC_NO,KC_NO,KC_NO,
+    KC_NO,KC_NO,KC_NO,KC_NO,KC_NO,KC_NO,KC_NO,
+    KC_NO,KC_NO,KC_NO,KC_MEDIA_PLAY_PAUSE,KC_NO,KC_NO,
+    KC_NO,KC_MEDIA_PREV_TRACK,KC_MEDIA_NEXT_TRACK,KC_NO,KC_NO,KC_AUDIO_VOL_UP,KC_AUDIO_VOL_DOWN,
+    KC_AUDIO_MUTE,KC_NO,KC_NO,KC_NO,KC_NO,
+    // BLOCK
+    KC_NO,KC_NO,KC_NO,KC_WWW_BACK),
+};
+
+const uint16_t PROGMEM fn_actions[] = {
+  [1] = ACTION_LAYER_TAP_TOGGLE(1)
+};
+
+// leaving this in place for compatibilty with old keymaps cloned and re-compiled.
+const macro_t *action_get_macro(keyrecord_t *record, uint8_t id, uint8_t opt)
```

Comment:
Could you remove the `action_get_macro` function?  It's been deprecated in favor of `process_record_user`.